src/base/static/components/input-form/index.js

```diff
@@ -273,10 +273,12 @@ class InputForm extends Component {
 
     return (
       <div className="input-form">
-        <WarningMessagesContainer
-          errors={[...this.state.formValidationErrors]}
-          headerMsg={this.props.t("validationHeader")}
-        />
+        {this.state.formValidationErrors.size && (
+          <WarningMessagesContainer
+            errors={[...this.state.formValidationErrors]}
+            headerMsg={this.props.t("validationHeader")}
+          />
+        )}
         <form
```

Comment:
I agree re: your advice to render/not render here instead of using a css modifier to control visibility. I believe we're still using the visibility pattern in the `Survey` component, but we could refactor when adding initial test(s) for that component I suppose.

Reply:
Eventually I think we should refactor WarningMessagesContainer itself to remove the `--visible` modifier and CSS logic to hide it by default.